product-scenarios/scenarios-commons/src/main/java/org/wso2/identity/scenarios/commons/util/SCIMProvisioningUtil.java

```diff
@@ -0,0 +1,116 @@
+        };
+        return headers;
+    }
+
+    /**
+     * Provision a new user
+     * @param serverURL              the server url
+     * @param jsonObject             the json object of the scim payload
+     * @param username               Authenticating username.
+     * @param password               Authenticating user password.
+     * @return HttpResponse with the result.
+     * @throws IOException If error occurs during app creation.
+     */
+    public static HttpResponse provisionUserSCIM11(String serverURL, JSONObject jsonObject,
+                                                   String username, String password) throws IOException {
+
+        return provisionSCIM11Entity(serverURL,Constants.SCIM_ENDPOINT_USER, jsonObject, username, password);
+    }
+
+
+    /**
+     * Provision a new user
+     * @param serverURL              the server url
+     * @param scim11EndpointType     type ex:User,Bulk,Group
+     * @param jsonObject             the json object of the scim payload
+     * @param username               Authenticating username.
+     * @param password               Authenticating user password.
+     * @return HttpResponse with the result.
+     * @throws IOException If error occurs during app creation.
+     */
+    public static HttpResponse provisionSCIM11Entity(String serverURL, String scim11EndpointType, JSONObject jsonObject,
+                                                   String username, String password) throws IOException {
+
+        HttpClient client = HttpClients.createDefault();
+        String scimEndpoint = serverURL + Constants.SCIM_11_ENDPOINT + "/" + scim11EndpointType;
+
+        return sendPostRequestWithJSON(client, scimEndpoint, jsonObject,getHeader(username, password));
+    }
+
+    /**
+     * Delete a  user
+     * @param serverURL              the server url.
+     * @param userId                 the ID of the user to be deleted.
+     * @param username               Authenticating username.
+     * @param password               Authenticating user password.
+     * @return HttpResponse with the result.
+     * @throws IOException If error occurs during app creation.
+     */
+    public static HttpResponse deleteUser(String serverURL,String userId,
+                                          String username, String password) throws IOException {
+
+        return deleteUserEntity(serverURL,userId,username, password);
+    }
+
+    /**
+     * Delete a  user
+     * @param serverURL              the server url
+     * @param userID                 the ID of the user to be deleted
+     * @param username               Authenticating username.
+     * @param password               Authenticating user password.
+     * @return HttpResponse with the result.
+     * @throws IOException If error occurs during app creation.
+     */
+    public static HttpResponse deleteUserEntity(String serverURL, String userID,
```

Comment:
Change this to `deleteSCIM11Entity`. Add an argument to get the endpoint type (Users/Groups) similar to `provisionSCIM11Entity`

Reply:
in both provisionSCIM11Entity and deleteSCIM11Entity a change was made where the endpoint type is not passed as a parameter but is taken from Constants class and passed to the URL when constructing the URL.

provisionSCIM11Entity
String scimEndpoint = serverURL + Constants.SCIM_11_ENDPOINT + "/" + Constants.SCIM_ENDPOINT_USER;

deleteSCIM11Entity
String scimEndpoint = serverURL + Constants.SCIM_11_ENDPOINT + "/" + Constants.SCIM_ENDPOINT_USER+ "/" + userID;

Reply:
Done, introduced this change now the test class needs to pass the serverURL and the SCIM endpoint type to the util class. From the scimEndPoint parameter we can pass ,wso2scim for scim 1.1 or scim2 for SCIM2.0 to construct the URL.